src/go/Rumble/rumble_server.go

```diff
@@ -0,0 +1,394 @@
+    releasePorts(port1, port2, portsInUseLockFname) // Release reserved ports
+    log.Println("[INFO] Ports released", port1, port2)
+    if !ok {
+        log.Println("[INFO] Fail.")
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(response)
+        return
+    }
+    //os.Remove(jsoniqFileName) // Remove temp JSONiq file
+    log.Println("[INFO] Success.")
+    w.WriteHeader(http.StatusOK)
+    _ = json.NewEncoder(w).Encode(response)
+}
+
+// RumbleServerRequestHandler we should only receive POST request
+func RumbleServerRequestHandler(w http.ResponseWriter, r *http.Request) {
+    w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+    w.Header().Set("Access-Control-Allow-Origin", "*")
+    w.Header().Set("Accept", "application/json")
+    w.Header().Set("Content-Type", "application/json")
+    var request Request
+    decoder := json.NewDecoder(r.Body)
+    err := decoder.Decode(&request)
+    log.Printf("[INFO] Request - Method: %v, Host: %v, RemoteAddr: %v, Args: %v, URI: %v",
+        r.Method, r.Host, r.RemoteAddr, r.URL, r.RequestURI)
+    if err != nil {
+        tempBody, _ := ioutil.ReadAll(r.Body)
+        log.Printf("[ERROR] could not get request body, reason: %v, Body:%#v", err, tempBody)
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(Response{
+            ErrorMsg: ErrorMessage{
+                Message:  "Please check your query, json could not be decoded!",
+                SysError: fmt.Sprintf("%v", err),
+            },
+        })
+        return
+    }
+    log.Printf("[INFO] Request Body:%#v", request)
+    // Check user parameters
+    if request.OutputPath == "" || request.UserId == "" || request.Query == "" {
+        w.WriteHeader(http.StatusBadRequest)
+        _ = json.NewEncoder(w).Encode(Response{
+            ErrorMsg: ErrorMessage{
+                Message: "'query', 'output_path' or 'userid' cannot be empty!",
+            },
+        })
+        return
+    }
+    // Set defaults
+    verbose := 0
+    if request.SparkConf.SparkExecutorMemory == "" {
+        request.SparkConf.SparkExecutorMemory = "2g"
+    }
+    if request.SparkConf.SparkExecutorInstances == "" {
+        request.SparkConf.SparkExecutorInstances = "2"
+    }
+    if request.SparkConf.SparkExecutorCores == "" {
+        request.SparkConf.SparkExecutorCores = "2"
+    }
+    if request.SparkConf.SparkDriverMemory == "" {
+        request.SparkConf.SparkDriverMemory = "2g"
+    }
+    processRumbleServerRequest(w, &request, verbose)
+}
+
+func handlers() *mux.Router {
+    router := mux.NewRouter()
+    fs := http.FileServer(http.Dir("./static"))
+    router.Handle("/", fs)
+    router.HandleFunc("/rumble-server", RumbleServerRequestHandler).Methods("POST", "OPTIONS")
+    return router
+}
+
+func main() {
+    portsInUseLockFname = "/var/lock/rumble-ports.json.lock"
+    initializePortsInUse(portsInUseLockFname)
```

Comment:
here your code will leave lock file in place after server will shutdown. To avoid that you need to use temp file such that it will be removed once your server is gone. Have a look how it is done in Go, e.g.
```
tmpFile, err := ioutil.TempFile(os.TempDir(), "prefix-")
```